lib/kafka_ex/protocol/common.ex

```diff
@@ -0,0 +1,16 @@
+defmodule KafkaEx.Protocol.Common do
+  @moduledoc """
+  A collection of common request generation and response parsing functions for the
+  Kafka wire protocol.
+  """
+
+  @doc """
+  Generate the wire representation for a list of topics.
+  """
+  def topic_data([]), do: ""
+
+  def topic_data([topic|topics]) do
```

Comment:
This is another educative read http://ferd.ca/erlang-s-tail-recursion-is-not-a-silver-bullet.html